tests/src/Integration/Action/EntityFetchByFieldTest.php

```diff
@@ -0,0 +1,209 @@
+        $this->typedDataManager,
+        $this->getMock('Drupal\Core\KeyValueStore\KeyValueStoreInterface'),
+        $this->getMock('Symfony\Component\EventDispatcher\EventDispatcherInterface')
+      ])
+      ->getMock();
+
+    // Return some dummy bundle information for now, so that the entity manager
+    // does not call out to the config entity system to get bundle information.
+    $this->entityManager->expects($this->any())
+      ->method('getBundleInfo')
+      ->with($this->anything())
+      ->willReturn(['entity_test' => ['label' => 'Entity Test']]);
+    $this->container->set('entity.manager', $this->entityManager);
+
+    $this->action = $this->actionManager->createInstance('rules_entity_fetch_by_field');
+  }
+
+  /**
+   * Tests the summary.
+   *
+   * @covers ::summary
+   */
+  public function testSummary() {
+    $this->assertEquals('Fetch entities by field', $this->action->summary());
+  }
+
+  /**
+   * Tests action execution when no value for limit is provided.
+   *
+   * @covers ::execute
+   */
+  public function testActionExecutionWithNoLimit() {
+    // Create variables for action context values.
+    $entity_type = 'entity_test';
+    $field_name = 'test_field';
+    $field_value = 'llama';
+
+    // Create an array of dummy entities.
+    $entities = [];
+    for ($i = 0; $i < 2; $i++) {
+      $entity = $this->getMock('Drupal\Core\Entity\EntityInterface');
+      $entities[] = $entity;
+    }
+
+    // Create dummy entity storage object.
+    $entityStorage = $this->getMock('Drupal\Core\Entity\EntityStorageInterface');
+    $entityStorage->expects($this->once())
+      ->method('loadByProperties')
+      ->with([$field_name => $field_value])
+      ->will($this->returnValue($entities));
+    $this->entityManager->expects($this->once())
+      ->method('getStorage')
+      ->with($entity_type)
+      ->will($this->returnValue($entityStorage));
+
+    // Set context values for EntityFetchByField action and execute.
+    $this->action->setContextValue('type', $entity_type)
+      ->setContextValue('field_name', $field_name)
+      ->setContextValue('field_value', $field_value)
+      ->execute();
+
+    // Test that executing action without a value for limit returns the dummy entities array.
+    $this->assertEquals($entities, $this->action->getProvidedContext('entity_fetched')->getContextValue('entity_fetched'));
+  }
+
+  /**
+   * Tests action execution when a value for limit is provided.
+   *
+   * @covers ::execute
+   */
+  public function testActionExecutionWithLimit() {
+    $entity_type = 'entity_test';
+    $field_name = 'test_field';
+    $field_value = 'llama';
+    $limit = 2;
+
+    // Create an array of dummy entities.
+    $entities = [];
+    for ($i = 0; $i < 4; $i++) {
+      $entity = $this->getMock('Drupal\Core\Entity\EntityInterface');
+      $entities[] = $entity;
+    }
+
+    // Create new dummy array of entities for testing limit.
+    $entities = array_slice($entities, 0, $limit);
```

Comment:
First you create 4 entities, but then you slice it down to 2, so the others are never used?